### PR TITLE
Main template

### PR DESCRIPTION
**Pull request by BalmungSan**
This depends on #5 and therefore should not be merged before syncing with that one.

### CODE REVIEW

File: src/cpp/main.cpp [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+  normal_forces = (double*) calloc(num_particles * num_particles, sizeof(double));
+  tangent_forces = (double*) calloc(num_particles * num_particles, sizeof(double));
+  forces = (Vector*) calloc(num_particles, sizeof(Vector));
+  accelerations = (Vector*) calloc(num_particles, sizeof(Vector));
+  velocities = (Vector*) calloc(num_particles, sizeof(Vector));
+  displacements = (Vector*) calloc(num_particles, sizeof(Vector));
+
+  // Initialize the structures.
+  for (size_t i = 0; i < num_particles; ++i) {
+    particles[i].radious = config->r;
```
Comment (BalmungSan) [inline src/cpp/main.cpp:56]: @ibotero2 How should we initialize the `x_coordinate` & `y_coordinate` properties of the particles _(including the falling one)_?

File: src/cpp/config.cpp [hunk 2]
```diff
@@ -0,0 +1,60 @@
+#include <fstream>
+#include <iostream>
+#include <sstream>
+#include <string>
+extern "C" {
+  #include "config.h"
+}
+
+using namespace std;
```
Comment (jandrovins) [inline src/cpp/config.cpp]: Including namespaces is a bad practice and I think we should avoid this. Our code should remain easily maintainable. If one day we happen to need something from other namespace then we should have to change lots of lines. [https://stackoverflow.com/questions/1452721/why-is-using-namespace-std-considered-bad-practice](url)

File: src/cpp/config.cpp [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+  string key;
+  string value;
+  while (getline(config_file, line)) {
+    // Skip empty lines.
+    if (line.empty()) {
+      continue;
+    }
+
+    // Parse each line.
+    istringstream is_line(line);
```
Comment (jandrovins) [inline src/cpp/config.cpp]: `iss_line` maybe? (input string stream line)

**Review by jandrovins — CHANGES_REQUESTED** [review]

**Comment by BalmungSan** [comment]
As you wish.

File: src/cpp/config.cpp (hunk 2 above)
Comment (BalmungSan) [reply src/cpp/config.cpp]: I would argue it is only a problem if we would use another namespace like boost.

Anyways, if you want you can make the changes on this branch yourself :+1:

File: src/cpp/config.cpp (hunk 3 above)
Comment (BalmungSan) [reply src/cpp/config.cpp]: As you wish.

File: src/cpp/config.cpp (hunk 2 above)
Comment (BalmungSan) [reply src/cpp/config.cpp]: Fixed in ae5ea3e3acb7a2f5c0fe74755608110cb96c3936.

File: src/cpp/config.cpp (hunk 3 above)
Comment (BalmungSan) [reply src/cpp/config.cpp]: Fixed in ae5ea3e3acb7a2f5c0fe74755608110cb96c3936.

**Comment by BalmungSan** [comment]
Guys, I believe these are enough changes for one PR.
I think we should merge this now _(unless you have any important suggestion)_ and any other changes like the simulation cage, should be done in other PRs.

Also, we are also missing the aesthetic changes, we should have an agreement about which standard are we going to follow, and if needed made the changes on another PR.

Looking forward to hearing from you.

**Review by jandrovins — APPROVED** [review]

**Pull request merged**